docs/pages/guides/extending-the-session.mdx

```diff
@@ -10,6 +10,31 @@ This is `name`, `email`, and `image`.
   a database or external APIs.
 </Callout>
 
+## With provider functions
+
+We can extend the default session data in a few ways, one of which is by using the `authorize` and `profile` functions. These functions let us return a user object with the properties we need. We can then create logic based on this information to search in a database or an external API.
+
+```ts
+import Github from "next-auth/providers/github";
+import Credentials from "next-auth/providers/credentials";
+import type { Provider } from "next-auth/providers";
+
+const providers: Provider[] = [
+  Google({
```

Comment:
You can drop the `clientId` and `clientSecret` values here, if the user sets the environment variables following the pattern we expect (for example `AUTH_GOOGLE_ID`/`AUTH_GOOGLE_SECRET`), Auth.js will automatically pick them up.

And regarding the profile callback, why do the extra step of spreading the `profile` argument into an object instead of just returning profile? Like this:

```js
Google({
    async profile(data) {
      return data
    },            
  }),
```